pkg/square/square_fuzz_test.go

```diff
@@ -70,3 +70,52 @@ func FuzzSquare(f *testing.F) {
 		}
 	})
 }
+
+// FuzzSquareDeconstruct tests whether square deconstruction function can correctly deconstruct a block back from a given square.
+func FuzzSquareDeconstruct(f *testing.F) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+
+	f.Add(0, 1)
+	f.Fuzz(func(t *testing.T, normalTxCount int, pfbCount int) {
+		// skip negative values
+		if normalTxCount < 0 || pfbCount < 0 {
+			t.Skip()
+		}
+		maxBlobSize := 1000 // @TODO there might be a global constant for this
+		allTxs := GenerateOrderedRandomTxs(encCfg.TxConfig, normalTxCount, pfbCount, maxBlobSize)
+
+		// extract those transaction that fit into the block
+		builtSquare, blockTxs, err := square.Build(allTxs, appconsts.LatestVersion, appconsts.DefaultSquareSizeUpperBound)
+		require.NoError(t, err)
+
+		// check that blockTxs is a subset of allTxs
+		require.True(t, contains(allTxs, blockTxs))
+
+		// construct the square
+		dataSquare, err := square.Construct(blockTxs, appconsts.LatestVersion, appconsts.DefaultSquareSizeUpperBound)
+		require.NoError(t, err)
+
+		require.Equal(t, builtSquare, dataSquare)
+
+		recomputedTxs, err := square.Deconstruct(dataSquare, encCfg.TxConfig.TxDecoder())
+		require.NoError(t, err)
+		require.Equal(t, len(blockTxs), len(recomputedTxs.ToSliceOfBytes()))
+		require.Equal(t, blockTxs, recomputedTxs.ToSliceOfBytes())
```

Comment:
Couldn't we just add these 4 lines to `FuzzSquare`

Reply:
We could, As I pointed out earlier, the two tests currently generate transactions in different ways. However, I can modify the `FuzzSquare` to use the `GenerateOrderedRandomTxs` if you think it's a good idea. What are your thoughts on this? @cmwaters

Reply:
Thanks for the clarification 🙏 
> All randomness should be deterministic (there should be no outside randomness). If the fuzzer outputs some seed corpus that has an unexpected result, we should be able to run it over and over again and expect to see the same failure.

I covered this point in this [comment](https://github.com/celestiaorg/celestia-app/pull/1860#discussion_r1224638076).

Reply:
> What randomness is meaningful and not supposed to be covered elsewhere? For example changing the signer is meaningless given that square construction only cares about arrays of bytes and blobs

I agree with your point.  the `generateRandomSigner` function is not intended to augment the randomness of the final generated transactions. As I stated in my comment [here](https://github.com/celestiaorg/celestia-app/pull/1860#discussion_r1223413788), the only additional elements of randomness introduced in this PR are 1) the blob counts for each blob transaction and 2) the amount for the send transaction. However, since the inclusion of  `generateRandomSigner` appears to be causing confusion, I will proceed with the removal of it.

Reply:
Done in https://github.com/celestiaorg/celestia-app/pull/1860/commits/bf6b0bf5c312bc7826e4b5c45c1c12b1dcdba5ff